syntax = "proto3";

package classification;

service ImageClassificationService {
  rpc Predict (ImageRequest) returns (ImageResponse);
}

message ImageRequest {
  bytes image_data = 1;  // raw image bytes
}

message Prediction {
  string class_name = 1;
  float confidence = 2;
}

message ImageResponse {
  repeated Prediction predictions = 1;
}
